edgekv/examples/mpulse-currency-converter-with-edgekv/main.js

```diff
@@ -0,0 +1,88 @@
+/*
+(c) Copyright 2020 Akamai Technologies, Inc. Licensed under Apache 2 license.
+Purpose:  Convert multiple currencies to one to support mPulse
+*/
+
+import { ReadableStream, WritableStream } from 'streams'
+import { httpRequest } from 'http-request'
+import { createResponse } from 'create-response'
+import { TextEncoderStream, TextDecoderStream } from 'text-encode-transform'
+import URLSearchParams from 'url-search-params';
+import { EdgeKV } from './edgekv.js';
+
+async function getCurrency(request){
+	//in this example the currency code from the purchase order is pulled from a query parameter
+    let params = new URLSearchParams(request.query);
+    let currencyCode   = params.get('currency'); 
+    let default_value  = 1;
+    
+    try {
+      const edgeKv = new EdgeKV({namespace: "namespace-name", group: "group-name"});
+      var currencyValue = await edgeKv.getText({item: currencyCode, default_value: default_value});
+        
+    } catch (error) {
+        return error.toString();

```

Comment:
I am not a JS expert by any means, but I suspect that with the current code, if you hit an exception (e.g. expired token, or some EdgeKV service failure), then "currencyValue" will not be defined.
I think you need to define it before the try/catch block, and you also need to give it a default value because if the exception is encountered, it will not be assigned a default value as is the case in here.

Reply:
btwm you can test my theory above by using an invalid token (e.g. modifying the token value in "edgekv_tokens.js", upload to staging and test with that. My guess is you'll get an EW execution error.

Reply:
good catch, tested, modified and works properly now